Add precision (by default) to Height/Weight Pickers

eschramm opened the pull request:
Closes https://github.com/CareEvolution/RKStudio-Participant/issues/1099. This is a simple approach to add precision mostly as discussed in https://github.com/CareEvolution/Consumers/issues/11101. 

ResearchKit currently does not have a high-precision concept for Height and for Weight and USC, it currently uses ounces.

While still allowing support for "low precision" (current behavior), the accompanying changes in [CEVResearchKit](https://github.com/CareEvolution/CEVResearchKit/pull/270) will now default to this new "high precision" mode.

|Mode|BEFORE|AFTER|
|---|---|---|
|**Height - USC**|![Height - USC - Low](https://user-images.githubusercontent.com/1008462/198047174-4e74302e-95b2-4fb5-9cd6-4563266b8543.png)|![Height - USC - High](https://user-images.githubusercontent.com/1008462/198047267-3bd45ff7-8d09-4b7c-8fb2-cceda3a2d302.png)|
|**Height - Metric**|![Height - Metric - Low](https://user-images.githubusercontent.com/1008462/198047465-21aabc8d-5af0-4ae4-862d-b44a4c3fadb1.png)|![Height - Metric - High](https://user-images.githubusercontent.com/1008462/198047512-0c07d1fb-59ea-4bc0-a495-d4a4cbb9e78f.png)|
|**Weight - USC**|![Weight - USC - Low](https://user-images.githubusercontent.com/1008462/198047620-7fc6d8c4-f7d3-495d-9e33-372e700d4091.png)|![Weight - USC - High](https://user-images.githubusercontent.com/1008462/198047651-f7aba0a8-dfb4-4b23-bfa4-52b6c0b30671.png)|
|**Weight - Metric**|![Weight-Metric-Low](https://user-images.githubusercontent.com/1008462/198047782-c4b52835-de6e-4d9c-b206-b4510a8c6984.png)|![Weight - Metric - High](https://user-images.githubusercontent.com/1008462/198047795-2ff8561a-d22b-4ff1-8da9-87b5e537ed2c.png)|

### Testing
- Check to see that for a **Question Step** or **Form Step** that uses an **Answer Format** of **Height**  shows ability to 
select fraction of inches as shown above
- Check to see that for a **Question Step** or **Form Step** that uses an **Answer Format** of **Weight**  shows ability to select fraction of pounds as shown above

eschramm committed:
add precision (by default) to Height/Weight Pickers

mmertsock commented on ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h:
```diff
@@ -1459,20 +1459,57 @@ ORK1_CLASS_AVAILABLE
+                                    default iOS behavior. If you pass `ORK1NumericPrecisionLow`, the
+                                    picker will use 1 cm increments for the metric measurement
+                                    system and whole pound increments for the USC measurement
+                                    system. If you pass `ORK1NumericPrecisionHigher`, the picker
```
Typo (here and various other places): "ORK1NumericPrecisionHigher" should be "ORK1NumericPrecisionHigh"

mmertsock commented on ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h:
```diff
@@ -1459,20 +1459,57 @@ ORK1_CLASS_AVAILABLE
+                                    increments for the USC measurement system, which mimics the
+                                    default iOS behavior. If you pass `ORK1NumericPrecisionLow`, the
+                                    picker will use 1 cm increments for the metric measurement
+                                    system and whole pound increments for the USC measurement
```
"whole pound increments"?

mmertsock commented on ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h:
```diff
@@ -1459,20 +1459,57 @@ ORK1_CLASS_AVAILABLE
+                                    use 0.1 cm increments for the metric measurement system,
+                                    and 0.1 inch increments for the USC measurement system.
+ 
+ @return An initialized weight answer format.
```
Typo: "weight" instead of "height"

mmertsock commented on ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.m:
```diff
@@ -2910,17 +2918,9 @@ - (NSString *)stringForAnswer:(id)answer {
-                NSString *ouncesString = [formatter stringFromNumber:@(ounces)];
-                answerString = [NSString stringWithFormat:@"%@ %@, %@ %@", poundsString, ORK1LocalizedString(@"MEASURING_UNIT_LB", nil), ouncesString, ORK1LocalizedString(@"MEASURING_UNIT_OZ", nil)];
-            }
+            double pounds = ORK1KilogramsToPounds(((NSNumber *)answer).doubleValue);
```
The `pounds` value here will [always be rounded to a whole integer](https://github.com/CareEvolution/ResearchKit/blob/8cfb94763af40d5fa208f3598104a84c7ee89117/ORK1Kit/ORK1Kit/Common/ORK1Helpers_Internal.h#L328) — not what we want, right?

eschramm replied:
Good catch, now that double from `ORK[1]KilogramsToPounds` is unrounded

mmertsock commented on ORK1Kit/ORK1Kit/Common/ORK1WeightPicker.m:
```diff
@@ -131,16 +131,16 @@ - (void)setAnswer:(id)answer {
         } else {
-            double pounds, ounces;
-            ORK1KilogramsToPoundsAndOunces(((NSNumber *)answer).doubleValue, &pounds, &ounces);
+            double pounds = ORK1KilogramsToPounds(((NSNumber *)answer).doubleValue);
```
Same comment here about ORK1KilogramsToPounds?

mmertsock reviewed (commented):
Overall design and implementation looks good, just a few questions/typos

eschramm committed:
PR fixes

- documentation type fixes - ORK1NumericPrecisionHigher -> ORK1NumericPrecisionHigh
- pound -> inch on height documentation
- weight -> height on height documentation
- fix kg -> lb conversion to be non-rounded double

eschramm committed:
PR fixes - RK2 equivalent

mmertsock reviewed (approved):
LGTM 🚢
